lib/date_time/date_time.c

```diff
@@ -29,12 +38,19 @@ LOG_MODULE_REGISTER(date_time, CONFIG_DATE_TIME_LOG_LEVEL);
 
 #if defined(CONFIG_DATE_TIME_MODEM)
 /* The magic number 115 corresponds to the year 2015 which is the default
- * year given by the modem when the cellular network has not pushed time
- * the modem.
+ * year given by the modem when the cellular network has not pushed time to
+ * the modem. 115 comes from the fact that POSIX time is delta from year 1900.
  */
 #define MODEM_TIME_DEFAULT 115
```

Comment:
This is used in the %XTIME handler. I don't think the year can ever be 2015 in that notification, so the check should be unnecessary. I don't know why it would be needed when asking for the time from modem either, because if time has not been set, modem should return an error (not year 2015).

Reply:
This was the case for earlier modem versions. No error was returned, but the time originated from 2015. I don't know if this is still the case, probably not.

Reply:
I tried that also 1.1.x branch MFW returns an error for AT+CCLK? when it does not have valid time. I'd say this check is not needed anymore.